site/content/en/docs/build-clusters.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "GKE Build Clusters"
+description: >
+  Sometimes you might want to run all of your jobs in a GKE cluster that is separate from Prow (perhaps running in your own GCP project). What you need is a dedicated "build cluster".
+weight: 3
+---
+
+## Overview
+
+By default Prow will schedule jobs in the same Kubernetes cluster as itself, but
+under the `test-pods` namespace. Setting up a separate build cluster will allow
+you to schedule jobs into a different Kubernetes cluster altogether. When
+everything's set up, Prow's component will schedule jobs into your build cluster
+(as shown in the [Arhictecture diagram](../overview/architecture)) instead of
+its own. This way, you can "bring your own build cluster" to Prow to make it
+scale however you see fit.
+
+## Running the build cluster setup script
+
+The Prow source repo comes with a default
+[create-build-cluster.sh](https://github.com/kubernetes/test-infra/blob/master/prow/create-build-cluster.sh)
+script which allows you to create a new GKE cluster with the intent of giving
+the [Kubernetes Prow instance](https://prow.k8s.io) access to it. Because there
+are different Prow instances and each instance has its own default settings
+(esp. for permissions), each instance has its own such script, forked from the
+default one. For example, [Google's OSS Prow instance](https://oss.gprow.dev)
+has its own script
+[here](https://github.com/GoogleCloudPlatform/oss-test-infra/blob/master/prow/oss/create-build-cluster.sh).
+
+The scripts all have prompts and ask you various questions to set everything up.
+
+## Making use of your new build cluster
+
+After you've provisioned your cluster, the next step is to try to run a job on
+it. There are two general requirements here:
+
+1. Jobs must be able to be scheduled by Prow into your build cluster.
+2. The jobs themselves must be able to upload artifacts to the GCS bucket used by
+[Deck](../components/deck).
+
+We look at both requirements below.
+
+### Let Prow schedule your jobs (actually use your build cluster)
+
+Prow is a Kubernetes cluster. So is your build cluster. In order for Prow to
+schedule jobs (i.e., create Kubernetes pods) into your build cluster, it must be
+able to authenticate as a KSA defined in your build cluster that has a
+[cluster-admin](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles)
+Kubernetes role. This way, Prow (with its various individual components) can
+freely create, update, and delete jobs (pods) in your build cluster as
+necessary.
+
+In order for Prow's components to authenticate as the cluster-admin KSA in your
+build cluster, they use a token. You can think of this token simply as a
+password, that when provided to your build cluster's API server along with the
+KSA name, grants authentication as this very same KSA.
+
+The question now turns to how we can generate this token. This can be done
+manually, but the token cannot be valid for longer than X days for security
```

Comment:
@chaodaiG Do you know how long this is? was it 2 days?

Reply:
2 days max

Reply:
To be more accurate, 2 days for GKE, the value might be different for other k8s providers